daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Util.scala

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package value
+
+import com.daml.lf.transaction.TransactionVersion
+import com.daml.lf.value.Value._
+
+object Util {
+
+  // equivalent to serialization + unserialization.
+  def normalize(
+      value0: Value[ContractId],
+      version: TransactionVersion,
+  ): Either[String, Value[ContractId]] =
+    try {
+      Right(assertNormalize(value0, version))
+    } catch {
+      case e: IllegalArgumentException => Left(e.getMessage)
+    }
+
+  @throws[IllegalArgumentException]
+  def assertNormalize(value0: Value[ContractId], version: TransactionVersion): Value[ContractId] = {
```

Comment:
Why not return an `Either` here? Performance? How big is the impact of that in practice?

Reply:
To write it in a functional way I would have to use traverse.
- I found traverse not very readable 
- I have to use scalaZ  
- I never remember which implicits, I have to import to use `traverseU`
- It uses more stack 

This is probably several time faster, but it should not have a big impact in practice.